.travis.yml

```diff
@@ -13,8 +13,9 @@ os:
   - windows
 
 before_script:
-  - rustup component add rustfmt
+  - rustup component add rustfmt clippy
 
 script:
   - cargo fmt -- --check
   - cargo test
+  - cargo clippy --all-targets
```

Comment:
I've just realised a couple of things after looking at the logs from the nightly CI runs.  This doesn't actually fail the build if a clippy warning is found.  We'd need to pass `-- --deny clippy::all` too in order to make the test effective.

Also, I've noticed that on nightly, we get a warning about a clippy lint being deprecated.  Should I change this script to only run rustfmt and clippy for the stable builds?  I'd update it to:
```bash
before_script:
  - if [[ ${TRAVIS_RUST_VERSION} == stable ]]; then
      rustup component add rustfmt clippy;
    fi

script:
  - if [[ ${TRAVIS_RUST_VERSION} == stable ]]; then
      cargo fmt -- --check;
    fi
  - cargo test
  - if [[ ${TRAVIS_RUST_VERSION} == stable ]]; then
      cargo clippy --all-targets -- --deny clippy::all;
    fi
```

Reply:
why did you change it to run `cargo fmt -- --check` only on the stable version?

I thought this should work on nightly as well...

Reply:
Just in case there was any difference between the two in the future.  Didn't want to end up trying to satisfy two conflicting styles.  I don't know likely or not that is.  I'd be fine leaving it set to run on all rust versions.

Reply:
Yes, I think leaving it as it is would be ok. Worst case we can change it in the future

Reply:
OK - I've left the rustfmt part as it was and only updated the clippy part.